client.go

```diff
@@ -110,6 +114,15 @@ type Client struct {
 	// ResponseLogHook allows a user-supplied function to be called
 	// with the response from each HTTP request executed.
 	ResponseLogHook ResponseLogHook
+
+	// CheckRetry specifies a policy for handling retries. It is called
+	// following each request with the response and error values returned by
+	// the http.Client. If CheckRetry returns false, the Client stops retrying
+	// and returns the response to the caller. If CheckRetry returns an error,
+	// that error value is returned in lieu of the error from the request. If
+	// an error is returned, it is up to the CheckResponse callback to properly
+	// close any response body before returning.
+	CheckRetry func(resp *http.Response, err error) (bool, error)
```

Comment:
Can we make this an interface similar to RequestLogHook and ResponseLogHook? It would be nice to have as part of the public API for any cases where you need to return one of these from some other function. That would also expose the description in the godoc so that the meaning of the `(bool, error)` is clear at a glance.